src/osd/PrimaryLogPG.cc

```diff
@@ -7727,14 +7728,20 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	bufferlist bl;
 	if (oi.is_omap()) {
 	  ObjectMap::ObjectMapIterator iter = osd->store->get_omap_iterator(
-	    ch, ghobject_t(soid)
+	    ch, ghobject_t(soid),
+	    // try to seek as many keys-at-once as possible for the sake of performance.
+	    // note complexity should be logarithmic, so seek(n/2) + seek(n/2) is worse
+	    // than just seek(n).
+	    filter_prefix >= start_after ? filter_prefix : start_after
```

Comment:
This is problematic.
when `filter_prefix` is to be applied, we want to seek *TO* the key (lower_bound),
when `start_after` suggests that we want to seek *AFTER* the key (upper_bound).

Reply:
Yes, that's the reason behind the naming distinction between `start_from` and `start_after`.